src/collections/componentSetBuilder.ts

```diff
@@ -168,13 +168,10 @@ export class ComponentSetBuilder {
       }
     }
 
-    if (apiversion) {
-      componentSet.apiVersion = apiversion;
-    }
-
-    if (sourceapiversion) {
-      componentSet.sourceApiVersion = sourceapiversion;
-    }
+    componentSet.apiVersion ??= apiversion;
+    componentSet.sourceApiVersion ??= sourceapiversion;
+    logger.debug(`ComponentSet apiVersion = ${componentSet.apiVersion}`);
+    logger.debug(`ComponentSet sourceApiVersion = ${componentSet.sourceApiVersion}`);
```

Comment:
checking my understanding...if I got a componentSet via componentSetBuilder, and used a manifest, then sourceApiVersion would have been set by CSB.

but not if the consumer passed in a sourceApiVersion -- then that would undo whatever the manifest version is?

So...would it make sense to disallow that behavior (like a clever union type on `ComponentSetOptions` that doesn't allow those props to co-exist)?  Or do you think there's a use case for it?

Reply:
The `sourceApiVersion` set by `ComponentSet.fromManifest()` takes precedence.  The `sourceApiVersion` passed in as an option to CSB would be ignored.